From 2ed3ff7977d05d367802eab40fbb1496c9c5b61f Mon Sep 17 00:00:00 2001
From: Alain Volmat <alain.volmat@foss.st.com>
Date: Thu, 23 Feb 2023 18:55:32 +0100
Subject: [PATCH 0015/1141] ARM: dts: stm32: enable DCMI DMA-MDMA chaining on
 stm32mp157d-ev1.dts

Enable the DMA-MDMA chaining for the dcmi (camera capture)
in order to be able to achieve higher resolution.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>
Change-Id: I7f9daf8e2a270a6829721046ee33c2d92a6dfb49
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/321081
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
---
 arch/arm/boot/dts/stm32mp157d-ev1.dts | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

--- a/arch/arm/boot/dts/stm32mp157d-ev1.dts
+++ b/arch/arm/boot/dts/stm32mp157d-ev1.dts
@@ -180,6 +180,14 @@
 	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&dcmi_pins_a>;
 	pinctrl-1 = <&dcmi_sleep_pins_a>;
+	/*
+	 * Enable DMA-MDMA chaining by adding a SRAM pool and
+	 * a MDMA channel
+	 */
+	sram = <&dcmi_pool>;
+
+	dmas = <&dmamux1 75 0x400 0x01>, <&mdma1 0 0x3 0x1200000a 0 0>;
+	dma-names = "tx", "mdma_tx";
 
 	port {
 		dcmi_0: endpoint {
@@ -657,6 +665,13 @@
 	status = "disabled";
 };
 
+&sram4 {
+	dcmi_pool: dcmi_pool@0 {
+		reg = <0x0 0x8000>;
+		pool;
+	};
+};
+
 &timers2 {
 	/* spare dmas for other usage (un-delete to enable pwm capture) */
 	/delete-property/dmas;
